reference/UiTPAS.v2.json

```diff
@@ -3053,6 +3244,128 @@
+          "Example": {
+            "name": "Lastname",
+            "firstName": "Firstname",
+            "cardSystem": {
+              "id": 27
+            },
+            "email": "firstname.lastname@example.com",
+            "dateOfBirth": "1980-01-01",
+            "postalCode": "9000",
+            "city": "Gent",
+            "registrationOrganizer": {
+              "id": "d2f898b1-455e-44ee-b08a-c4c05b1ab900"
+            }
+          }
+        },
+        "description": "",
+        "x-tags": [
+          "Models"
+        ],
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "firstName": {
+            "type": "string"
+          },
+          "lastName": {
+            "type": "string"
+          },
+          "dateOfBirth": {
+            "type": "string",
+            "format": "date"
+          },
+          "inszNumber": {
+            "type": "string"
+          },
+          "address": {
+            "type": "object",
+            "required": [
+              "street",
+              "number",
+              "box",
+              "postalCode",
+              "city"
+            ],
+            "properties": {
+              "street": {
+                "type": "string"
+              },
+              "number": {
+                "type": "string"
+              },
+              "box": {
+                "type": "string"
+              },
+              "postalCode": {
+                "type": "string"
+              },
+              "city": {
+                "type": "string"
+              }
+            }
+          },
+          "email": {
+            "type": "string"
+          },
+          "cardSystem": {
+            "$ref": "#/components/schemas/CardSystem"
+          },
+          "registrationOrganizer": {
+            "$ref": "#/components/schemas/Organizer"
+          },
+          "gender": {
+            "type": "string",
+            "enum": [
+              "male",
+              "female",
+              "x"
+            ]
+          },
+          "nationality": {
+            "type": "string"
+          },
+          "phoneNumber": {
+            "type": "string"
+          },
+          "optInPreferences": {
+            "type": "object",
+            "properties": {
+              "serviceMails": {
+                "type": "boolean"
+              },
+              "milestoneMails": {
+                "type": "boolean"
+              },
+              "infoMails": {
+                "type": "boolean"
+              },
+              "sms": {
+                "type": "boolean"
+              },
+              "post": {
+                "type": "boolean"
+              }
+            }
+          }
+        },
+        "required": [
+          "name",
+          "firstName",
+          "lastName",
+          "dateOfBirth",
+          "inszNumber",
+          "address",
+          "cardSystem",
+          "registrationOrganizer"
+        ]
```

Comment:
I think you can use the same passholder schema both for registering a new passholder and retrieving a passholder.

Aside from some minor inconsistencies that should probably be fixed anyway (like `lastName` being required for registration but not when retrieving, and `gender` being a property for registration but not when retrieving) the only real "issue" is `cardSystem` vs `cardSystemMemberships`. But you can set `cardSystemMemberships` to `readOnly` and `cardSystem` to `writeOnly` and then it should work for both cases?

Reply:
Technically you can do it with 2 schemas of course but I think they will get out of sync really fast, seeing how there are already some inconsistencies here and there.